### PR TITLE
Make it easier to bump the metadata version

### PR DESCRIPTION
**Pull request by chrisccoulson**
This splits keyData, keyPolicyUpdateData, staticPolicyData and
dynamicPolicyData in to version agnostic parts and version-specific
parts. The version-specific parts are what is marshalled / unmarshalled.

### CODE REVIEW

File: policy.go [hunk 1]
```diff
@@ -67,6 +67,16 @@ type policyOrDataNode struct {
 
 type policyOrDataTree []policyOrDataNode
 
+type dynamicPolicyDataRaw_v0 dynamicPolicyData
```
Comment (pedronis) [inline policy.go:70]: the order of the definitions here is a bit odd,  I think maybe a comment could help about the relation of _v0 vs dynamicPolicyData atm

same for static

File: policy.go [hunk 2]
```diff
@@ -67,6 +67,16 @@ type policyOrDataNode struct {
...
+
+func (d *dynamicPolicyDataRaw_v0) data() *dynamicPolicyData {
+	return (*dynamicPolicyData)(d)
+}
+
+func makeDynamicPolicyDataRaw_v0(data *dynamicPolicyData) *dynamicPolicyDataRaw_v0 {
```
Comment (pedronis) [inline policy.go:76]: this name puzzled me a bit, but I suppose in general this wouldn't be a cast but produce a different *struct, right? so "make" makes sense

File: policy.go [hunk 3]
```diff
@@ -565,7 +585,10 @@ func computePCRSelectionListFromValues(v tpm2.PCRValues) (out tpm2.PCRSelectionL
 
 // computeDynamicPolicy computes the part of an authorization policy associated with a sealed key object that can change and be
 // updated.
-func computeDynamicPolicy(alg tpm2.HashAlgorithmId, input *dynamicPolicyComputeParams) (*dynamicPolicyData, error) {
+func computeDynamicPolicy(version uint32, alg tpm2.HashAlgorithmId, input *dynamicPolicyComputeParams) (*dynamicPolicyData, error) {
+	if version != 0 {
```
Comment (pedronis) [inline policy.go:589]: this probably merits a comment, because I suppose it is intentional that 0 is hardcoded

File: policy.go (hunk 2 above)
Comment (chrisccoulson) [reply policy.go:76]: Yeah, I did it this way because once the metadata format is changed, this will do something other than cast.

**Pull request merged**